standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5563,26 +5563,21 @@ public List<Partition> get_partitions_with_auth(final String dbName,
   private void checkLimitNumberOfPartitionsByFilter(String catName, String dbName,
                                                     String tblName, String filterString,
                                                     int maxParts) throws TException {
-    if (isPartitionLimitEnabled()) {
+    if (needCheckPartitionLimit(maxParts)) {
```

Comment:
why do we need `needCheckPartitionLimit` since we have the same checks in `checkLimitNumberOfPartitions`

Reply:
It's different. `needCheckPartitionLimit` use maxParts to determine if need run the `checkLimitNumberOfPartitions`, while `checkLimitNumberOfPartitions` checks if  the matching `numPartitions` is greater than limit.

Reply:
1. we execute the same function before and after db fetch - 'needCheckPartitionLimit', maybe we could rename and reuse 
2. also why we check count and fail post db call and not limit cound inside of it?

Reply:
````
  private void checkLimitNumberOfPartitionsByPs(String catName, String dbName, String tblName,
                                                List<String> partVals, int maxParts)
          throws TException {
    if (needCheckPartitionLimit(maxParts)) {
      checkLimitNumberOfPartitions(tblName, getNumPartitionsByPs(catName, dbName, tblName,
              partVals));
    }
  }

  private boolean needCheckPartitionLimit(int requestSize) {
    int partitionLimit = MetastoreConf.getIntVar(conf, ConfVars.LIMIT_PARTITION_REQUEST);
    return partitionLimit > -1 && (requestSize < 0 || requestSize > partitionLimit);
  }

  private void checkLimitNumberOfPartitions(String tblName, int numPartitions) throws MetaException {
    int partitionLimit = MetastoreConf.getIntVar(conf, ConfVars.LIMIT_PARTITION_REQUEST);
    if (partitionLimit > -1 && numPartitions > partitionLimit) {
      String configName = ConfVars.LIMIT_PARTITION_REQUEST.toString();
      throw new MetaException(String.format(PARTITION_NUMBER_EXCEED_LIMIT_MSG, numPartitions,
          tblName, partitionLimit, configName));
    }
  }
````
where do we limit the fetch size? we throws exception if returned `numPartitions` > `partitionLimit`?

Reply:
1. `needCheckPartitionLimit` is used to determine if needs to check liimt.
let's say that a request size is 100, LIMIT is 50, then the returned partition size may hit limit, so we need check limit;
Another request size is 30, LIMIT is 50, then returned partition size is never greater than limit, so we do not need check limit.
Let's rename it to `exceedsPartitionLimit(int size)` to reuse it.
2. The existing behavior is throw exception when hit partition limit, I think it's reasonable because returning the limit size partitions is confused whether there are other partitions matching my condition and it maybe not the expected result to users.

Reply:
1. sounds good
2. makes sense, but do we return an error back to the client saying that the configured partition fetch limit was exceeded or just log an exception?